lib/commands/index.js

```diff
@@ -132,24 +132,45 @@ const playback = {
         const response = map.getResponse(id, req, options);
         if (response) {
           try {
-            // We need to delete the 'content-encoding' header, as the recorded
-            // response will not have the body in that format due to compression.
-            // Notes on 'content-encoding':
-            // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding
-            delete response.headers['content-encoding'];
-            // Any headers that contain an array need to be either flattened
-            // if they only have one entry or deleted as Cypress does not
-            // handle array headers in response:
-            // 'An invalid StaticResponse was supplied to req.reply(). headers must be a map of strings to strings.'
-            for (const [key, value] of Object.entries(response.headers)){
-              if(Array.isArray(value) && value.length === 1) {
-                console.log(`Flattening response header "${key}"`);
-                response.headers[key] = value[0];
-              } else if(Array.isArray(value) && value.length >= 1) {
-                delete response.headers[key];
-                console.warn(`The response header "${key}" is an Array with multiple entries and was deleted.`);
-              }
-            }
+            // Update the headers so they work correctly when being played back.
+            response.headers = Object.fromEntries(Object.entries(response.headers).reduce(
```

Comment:
wondering if there’s something i don’t understand here — i would think you could make the accumulator an object and skip wrapping it in `Object.fromEntries`. would that not work? or would it be less readable?

Reply:
It probably does come down to style, I don't know if there's any performance difference. I think we'd have something like this with a new object.

```JavaScript
const newHeaders = {};
for(const [key, value] of Object.entries(response.headers)) {
  // ... some lines where we have `newHeaders[key] = value` or something.
}
response.headers = newHeaders;
```

Reply:
i was talking about like this:

```javascript
response.headers = Object.entries(response.headers).reduce(
  (acc, [key, value]) => {
    switch (key.toLowerCase()) {
      case 'content-encoding':
        // We need to remove the 'content-encoding' header, as the
        // recorded response will not have the body in that format
        // due to compression. Notes on 'content-encoding':
        // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding
        break;
      case 'access-control-allow-origin': {
        // We may not be playing back on the same origin that we
        // recorded, so update to the current origin.
        acc[key] = window.origin
        break;
      }
      default: {
        // Any headers that contain an array need to be either flattened
        // if they only have one entry or deleted as Cypress does not
        // handle array headers in response:
        // 'An invalid StaticResponse was supplied to req.reply(). headers must be a map of strings to strings.'
        if (Array.isArray(value)) {
          if (value.length === 1) {
            console.log('CYPRESS PLAYBACK:', `Flattening response header "${key}"`);
            acc[key] = value[0];
          } else {
            console.warn('CYPRESS PLAYBACK:', `The response header "${key}" is an Array with multiple entries and was deleted.`);
          }
        } else {
          acc[key] = value;
        }
        break;
      }
    }
    return acc;
  },
  {}
);
```

one less mental transformation to figure out what the result looks like